src/System.IO.Compression/ref/System.IO.Compression.cs

```diff
@@ -58,14 +58,14 @@ public partial class GZipStream : System.IO.Stream
         public GZipStream(System.IO.Stream stream, System.IO.Compression.CompressionLevel compressionLevel, bool leaveOpen) { }
         public GZipStream(System.IO.Stream stream, System.IO.Compression.CompressionMode mode) { }
         public GZipStream(System.IO.Stream stream, System.IO.Compression.CompressionMode mode, bool leaveOpen) { }
-        public System.IO.Stream BaseStream { get { throw null; } }
+        public System.IO.Stream? BaseStream { get { throw null; } }
```

Comment:
On BufferedStream we decided as part of API review that its UnderlyingStream should be annotated to return `Stream` rather than `Stream?`, even though it could end up returning null if accessed after the stream was disposed.  That's because using the stream after it's been disposed is considered user error / misuse and the stream is invalid at that point, so it's ok that it's not abiding by the nullable contract.  We should follow suit here, assuming it can only be null after Dispose.

Reply:
Yes it will be null only if the stream is disposed, will update